dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java

```diff
@@ -56,71 +54,57 @@ public static void premain(final String arg, final Instrumentation inst) throws
-                            // .ignore(nameEndsWith("LocalTransactionAndCloseDBIfOpenedFactoryTest"))
-                            // .ignore(nameEndsWith("FolderAPITest"))
-                            .disableClassFormatChanges().with(AgentBuilder.InitializationStrategy.Minimal.INSTANCE)
-                            .with(AgentBuilder.TypeStrategy.Default.REDEFINE)
-                            .with(AgentBuilder.RedefinitionStrategy.RETRANSFORMATION);
+            AgentBuilder.RedefinitionListenable.WithoutBatchStrategy builder = new AgentBuilder.Default()
+                    .with(bootFallbackLocationStrategy)
+                    // Filtering by package name is quicker than by annotation
+                    .ignore(nameStartsWith("net.bytebuddy."))
+                    .ignore(nameStartsWith("com.dotcms.repackage."))
+                    .ignore(nameStartsWith("com.dotcms.business.bytebuddy"))
+                    .disableClassFormatChanges()
+                    .with(AgentBuilder.InitializationStrategy.Minimal.INSTANCE)
+                    .with(AgentBuilder.TypeStrategy.Default.REDEFINE)
+                    .with(AgentBuilder.RedefinitionStrategy.RETRANSFORMATION);
 
             /*
              * We do not use Config class for this as we do not want to load classes that may need
              * transformation or require annotation processing.
              */
             if (Boolean.parseBoolean(System.getenv("DOTCMS_BYTEBUDDY_DEBUG"))) {
-                builder = ((AgentBuilder.RedefinitionListenable) builder)
-                                .with(AgentBuilder.RedefinitionStrategy.Listener.StreamWriting.toSystemError())
-                                .with(AgentBuilder.Listener.StreamWriting.toSystemError().withTransformationsOnly())
-                                .with(AgentBuilder.InstallationListener.StreamWriting.toSystemError());
+                builder.with(AgentBuilder.RedefinitionStrategy.Listener.StreamWriting.toSystemError())
+                        .with(AgentBuilder.Listener.StreamWriting.toSystemError().withTransformationsOnly())
+                        .with(AgentBuilder.InstallationListener.StreamWriting.toSystemError());
             }
 
-            builder.with(bootFallbackLocationStrategy).type(ElementMatchers.nameStartsWith("com.dotcms"))
-                            .or(ElementMatchers.nameStartsWith("com.dotmarketing"))
-                            .or(ElementMatchers.nameStartsWith("com.liferay"))
-                            .transform(new AgentBuilder.Transformer.ForAdvice()
-                                            .withExceptionHandler(Advice.ExceptionHandler.Default.RETHROWING)
-                                            .advice(isMethod().and(isAnnotatedWith(CloseDB.class)),
-                                                            CloseDBAdvice.class.getName()))
+            builder.type(ElementMatchers.declaresMethod(
```

Comment:
should we have both sets of filters - package names AND class names for even faster startup?

Reply:
@wezell I have specific excludes of some things we know we want to ignore.  It is possible we would speed up a bit if we add more filters on name/package, there is no real different with class name and package name with this respect it is just using the name passed and at these ignores no reflection is being done and adding individual classes would get unwieldy.     If it gets through the ignores it my take slightly longer to check the isAnnotatedWith but this check is quicker than trying to transform the class.   We could use a whitelist on name but I am concerned that it may cause us problems later and maybe with plugins.   It seems to be quick starting up with this change, but we could possibly improve.   One way could be using what Quarkus uses called Jandex which creates an index of classes, annotations and generic types at build time and you can query it very quickly at runtime without reflection.  We may not even have the usual reflection hit here anyway as the class is being redefined at the point of initial loading so it actually has all the info it needs at that point and once transformed does not need to use reflection anymore.  I  Don't want to go too far on solving the performance here at the moment and add complexity if it is not making any difference but we do have options.

`.ignore(nameStartsWith("net.bytebuddy."))
                    .ignore(nameStartsWith("com.dotcms.repackage."))
                    .ignore(nameStartsWith("com.dotcms.business.bytebuddy"))`

Reply:
@wezell I have added a commit that does do the extra whitelist filtering as we had before and improved the logging; we could add config to this if we need to.